src/components/VideoPlayer/index.js

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable no-underscore-dangle */
+import {ResizeMode, Video} from 'expo-av';
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {View} from 'react-native';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import {usePlaybackContext} from '@components/VideoPlayerContexts/PlaybackContext';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import addEncryptedAuthTokenToURL from '@libs/addEncryptedAuthTokenToURL';
+import stylePropTypes from '@styles/stylePropTypes';
+import styles from '@styles/styles';
+import VideoPlayerControls from './VideoPlayerControls';
+
+const propTypes = {
+    url: PropTypes.string.isRequired,
+
+    shouldPlay: PropTypes.bool,
+
+    onVideoLoaded: PropTypes.func,
+
+    resizeMode: PropTypes.string,
+
+    isLooping: PropTypes.bool,
+
+    style: stylePropTypes,
+
+    videoStyle: stylePropTypes,
+
+    shouldUseSharedVideoElement: PropTypes.bool,
+};
+
+const defaultProps = {
+    shouldPlay: false,
+    onVideoLoaded: () => {},
+    resizeMode: ResizeMode.CONTAIN,
+    isLooping: false,
+    style: [styles.w100, styles.h100],
+    videoStyle: [styles.w100, styles.h100],
+    shouldUseSharedVideoElement: false,
+};
+
+function VideoPlayer({url, resizeMode, shouldPlay, onVideoLoaded, isLooping, style, videoStyle, shouldUseSharedVideoElement}) {
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const {currentlyPlayingURL, updateSharedElements, sharedElement, originalParent, updateCurrentVideoPlayerRef, currentVideoPlayerRef, updateIsPlaying} = usePlaybackContext();
+    const [duration, setDuration] = useState(0);
+    const [position, setPosition] = useState(0);
+    const [isLoading, setIsLoading] = useState(true);
+    const videoPlayerRef = useRef(null);
+    const videoPlayerElementParentRef = useRef(null);
+    const videoPlayerElementRef = useRef(null);
+    const sharedVideoPlayerParentRef = useRef(null);
+    const sourceURLWithAuth = addEncryptedAuthTokenToURL(url);
+
+    const onReadyForDisplay = useCallback(
+        (e) => {
+            onVideoLoaded(e);
+        },
+        [onVideoLoaded],
+    );
```

Comment:
Would it work?
```suggestion
    const onReadyForDisplay = useCallback(onVideoLoaded, [onVideoLoaded]);
```

And do we even need to memoize that?

I think we could just use this `onVideoLoaded` function directly, as this `useCallback` will always return the `onVideoLoaded` function, and will update every time `onVideoLoaded` got updated.